### PR TITLE
Implement Wildfire Druid: Enhanced Bond

### PR DESCRIPTION
**Pull request by Aeristoka**
https://www.dndbeyond.com/classes/druid#CircleofWildfireUA

Enhanced Bond
6th-level Circle of Wildfire feature

The bond with your wildfire spirit enhances your destructive and restorative spells. Whenever you cast a spell that deals fire damage or restores hit points while your wildfire spirit is summoned, roll a d8, and you gain a bonus to one roll of the spell equal to the number rolled.

### CODE REVIEW

**Comment by kakaroto** [comment]
```js
        if (character.hasClassFeature("Enhanced Bond") &&
            character.getSetting("wildfire-spirit-enhanced-bond", false) &&
            damages.length > 0) {
            for (let i = 0; i < damages.length; i++){
                if (damage_types[i] === "Fire") {
                    damages.push("1d8");
                    damage_types.push("Enhanced Bond");
                    break;
                }
            }
        }
```
Better way to do it (IMHO) :
```
        if (character.hasClassFeature("Enhanced Bond") &&
            character.getSetting("wildfire-spirit-enhanced-bond", false) &&
            damage_types.some(d => d === "Fire")) {
              damages.push("1d8");
              damage_types.push("Enhanced Bond");
        }
```

**Comment by Aeristoka** [comment]
> ```js
>         if (character.hasClassFeature("Enhanced Bond") &&
>             character.getSetting("wildfire-spirit-enhanced-bond", false) &&
>             damages.length > 0) {
>             for (let i = 0; i < damages.length; i++){
>                 if (damage_types[i] === "Fire") {
>                     damages.push("1d8");
>                     damage_types.push("Enhanced Bond");
>                     break;
>                 }
>             }
>         }
> ```
> 
> Better way to do it (IMHO) :
> 
> ```js
>         if (character.hasClassFeature("Enhanced Bond") &&
>             character.getSetting("wildfire-spirit-enhanced-bond", false) &&
>             damage_types.some(d => d === "Fire")) {
>               damages.push("1d8");
>               damage_types.push("Enhanced Bond");
>         }
> ```

Witchcraft!

I learn more JS from you all the time.

File: dist/dndbeyond_character.js [hunk 1]
```diff
@@ -2346,7 +2364,15 @@ class Beyond20RollRenderer {
                     critical_damage_rolls.push(roll);
                     const dmg_type = critical_damage_types[i];
                     let damage_flags = DAMAGE_FLAGS.REGULAR;
+<<<<<<< HEAD
+<<<<<<< HEAD
                     if (["Healing", "Disciple of Life", "Temp HP", "Alchemical Savant Healing"].includes(dmg_type)) {
+=======
+=======
```
Comment (kakaroto) [inline dist/dndbeyond_character.js:2371]: I particularly like this change... the conflict is conflicting with the conflict...
image

**Comment by kakaroto** [comment]
Merged by https://github.com/kakaroto/Beyond20/commit/a7297da8b4faba3c76907aae5fb759f9d900c755

**Pull request closed without merging**